sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -206,15 +207,40 @@ def _infer_event_timestamp_from_dataframe(entity_df: pandas.DataFrame) -> str:
 
 
 class BigQueryRetrievalJob(RetrievalJob):
-    def __init__(self, query, client):
+    def __init__(self, query, client, config):
         self.query = query
         self.client = client
+        self.config = config
 
     def to_df(self):
         # TODO: Ideally only start this job when the user runs "get_historical_features", not when they run to_df()
         df = self.client.query(self.query).to_dataframe(create_bqstorage_client=True)
         return df
 
+    def to_bigquery(self, dry_run=False) -> str:
+        today = date.today().strftime("%Y%m%d")
+        rand_id = str(uuid.uuid4())[:7]
+        path = f"{self.client.project}.{self.config.offline_store.dataset}.training_{today}_{rand_id} "
+
+        job_config = bigquery.QueryJobConfig(destination=path, dry_run=dry_run)
+        bq_job = self.client.query(self.query, job_config=job_config)
+
+        if dry_run is True:
+            print(
+                "This query will process {} bytes.".format(bq_job.total_bytes_processed)
+            )
+            return path
```

Comment:
Should we just return nothing when we asked for a dry_run ? I don't understand the utility of the `path` if the table never been created. What I am missing 🤔 ?

```python
if dry_run:
    return

...
return path
```

It just means that this function will output an Optional[str]
```python
def to_bigquery(self, dry_run=False) -> Optional[str]:
```

Reply:
Good point :) changed :D